project-utils/clock.test.js

```diff
@@ -0,0 +1,57 @@
+const moment = require('moment');
+const utils = require('./clock-utils');
+
+describe('clock helper: get seconds to', () => {
+  it('should return a number', () => {
+    const formatted = moment().add(1000, 'seconds').format('YYYY[-]MM[-]DD[ ]HH[:]mm[:]ss');
+    const diff = utils.getSecondsTo(formatted);
+    expect(typeof diff).toBe('number');
+  });
+
+  it('should return 999 when given a time 1000 seconds from now', () => {
+    const formatted = moment().add(1000, 'seconds').format('YYYY[-]MM[-]DD[ ]HH[:]mm[:]ss');
+    const diff = utils.getSecondsTo(formatted);
+    expect(diff).toBe(999);
+  });
+
+  it('should accept a date formatted YYYY[-]MM[-]DD[ ]HH[:]mm[:]ss as input', () => {
+    const formatted = moment().add(1000, 'seconds').format('YYYY[-]MM[-]DD[ ]HH[:]mm[:]ss');
+    const diff = utils.getSecondsTo(formatted);
+    expect(diff).not.toBeFalsy();
+  });
+
+  it('should not accept non-date values as input', () => {
+    const diff = utils.getSecondsTo('apple');
+    expect(diff).not.toBeTruthy();
+  });
+});
```

Comment:
I like this, but I might add another test for entries which are date-like but not in the correct format.

Reply:
I thought about that. I'm not going to modify the moment library for this, so given that I'm controlling this function and every invocation of it, all I want is to confirm that the way I will be inputting dates works and that non-dates cause a problem.